test/utils/unit.go

```diff
@@ -237,3 +238,33 @@ func NewBareMetalHost(namespace, name string) *v1alpha1.BareMetalHost {
 		},
 	}
 }
+
+func NewOpenshiftAssistedConfigWithInfraEnv(
+	namespace, name, clusterName string,
+	infraEnv *v1beta1.InfraEnv,
+) *v1alpha2.OpenshiftAssistedConfig {
+	var ref *corev1.ObjectReference
+	if infraEnv != nil {
+		ref = &corev1.ObjectReference{
+			Namespace: infraEnv.GetNamespace(),
+			Name:      infraEnv.GetName(),
+		}
+	}
```

Comment:
Do we need this check? Couldn't we just set the InfraEnvRef to whatever's passed in even if it's nil?